include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -0,0 +1,134 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_COMMON_H
+#define _ONEDPL_PSTL_OFFLOAD_INTERNAL_USM_MEMORY_REPLACEMENT_COMMON_H
+
+#include <sycl/sycl.hpp>
+#include <cstddef>
+#include <cstdlib>
+#include <cassert>
+#include <cstring>
+#include <limits>
+
+#if __linux__
+#include <dlfcn.h>
+#include <unistd.h>
+#endif // __linux__
+
+namespace __pstl_offload {
+
+inline constexpr std::size_t __uniq_type_const = 0x23499abc405a9bccLLU;
+inline constexpr std::size_t __header_offset = 32;
+
+struct alignas(__header_offset) __block_header {
+    std::size_t _M_uniq_const;
+    void* _M_original_pointer;
+    sycl::device* _M_device;
+    std::size_t _M_requested_number_of_bytes;
+}; // struct __block_header
+
+static_assert(sizeof(__block_header) <= __header_offset);
+
+#if __linux__
+
+inline constexpr bool __is_power_of_two(std::size_t __number) {
+    return __number != 0 && (__number & __number - 1) == 0;
```

Comment:
please add parentheses for the sake of readability

Reply:
Done